packages/process/src/node/terminal-process.ts

```diff
@@ -226,6 +238,19 @@ export class TerminalProcess extends Process {
                 }
             });
 
+            /**
+             * Wait for all the streams to be closed, as well as for the process
+             * to exit.
+             */
+            Promise.all([onExit, onClose]).then(resolved => {
```

Comment:
I'm not sure whether we need to do it, or we can simply translate `exit` and `close` events similarly to raw process implementation. Do you think semantic on node-pty `close` and `exit` is different from Node.js runtime or they try to emulate it as well?

Reply:
I am not sure, so I prefer not rely on whatever they do and rather make sure it happens in the right order here.

Reply:
ok, asked here: https://github.com/microsoft/node-pty/issues/369

Reply:
And also yes, the close event comes from the underlying socket object, at which point it doesn't really knows any exit code or signal, so we have to get it from the exit event.

Reply:
Or maybe we don't need  `code` and `signal` for close as the common denominator between Node-pty and child_processes. Then we can always simply translate.

Looking at unix implementation: https://github.com/microsoft/node-pty/blob/97b3ca171a05e01491150a553dc4f31c0d74203c/src/unixTerminal.ts#L74-L99

`exit` always comes after `close`, but `close` comes without `code` and `signal` as you said.

Reply:
I think in the case of node-pty, the `close` event simply refers to the underlying socket object used to handle io streams: doing `.write(...)` or `.on('data', ...)` actually goes through the socket. So when the socket emits `close`, terminals will bubble that event up themselves, and it will mean that io is closed, just like Node's child processes.

The difference seems to lie in the code and signal parameters not being passed, and also the events order seems weird: To me, we should first get `exit` and `close` after, hence why I am waiting for both `onExit` and `onClose` before actually firing the `close` event from our processes.

Reply:
ok